plan/physical_plan_builder.go

```diff
@@ -731,16 +730,31 @@ func (p *DataSource) forceToIndexScan(idx *model.IndexInfo) PhysicalPlan {
 	return t.plan()
 }
 
+// If there is a table reader which needs to keep order, we should append a pk to table scan.
+func (ts *PhysicalTableScan) appendPK(ds *DataSource) {
```

Comment:
The method name is not consistent with its behavior.
I think splitting it into two methods `hasHandleCol` and `appendExtraHandleCol` would be more clear.